docs/reference/http.rst

```diff
@@ -0,0 +1,218 @@
+
+Retrieve instance metrics.
+
+.. code-block::
+
+    http://<hostname>:<port>/metrics
+
+All EdgeDB instances expose a Prometheus-compatible endpoint available via GET
+request. The following metrics are made available.
+
+Processes
+^^^^^^^^^
+
+``compiler_process_spawns_total``
+  **Counter.** Total number of compiler processes spawned.
+
+``compiler_processes_current``
+  **Gauge.** Current number of active compiler processes.
+
+Backend connections and performance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+``backend_connections_total``
+  **Counter.**
+  Total number of backend connections established.
+
+``backend_connections_current``
+  **Gauge.** Current number of active backend connections.
+
+``backend_connection_establishment_errors_total``
+  **Counter.** Number of times the server could not establish a backend
+  connection.
+
+``backend_connection_establishment_latency``
+  **Histogram.** Time it takes to establish a backend connection, in seconds.
+
+``backend_query_duration``
+  **Histogram.** Time it takes to run a query on a backend connection, in
+  seconds.
+
+Client connections
+^^^^^^^^^^^^^^^^^^
+
+``client_connections_total``
+  **Counter.** Total number of clients.
+
+``client_connections_current``
+  **Gauge.** Current number of active clients.
+
+``client_connections_idle_total``
+  **Counter.** Total number of forcefully closed idle client connections.
+
+Query compilation
+^^^^^^^^^^^^^^^^^
+
+``edgeql_query_compilations_total``
+  **Counter.** Number of compiled/cached queries or scripts.
+
+``edgeql_query_compilation_duration``
+  **Histogram.** Time it takes to compile an EdgeQL query or script, in
+  seconds.
+
+Errors
+^^^^^^
+
+``background_errors_total``
+  **Counter.** Number of unhandled errors in background server routines.
+
+.. _ref_reference_http_querying:
+
+Querying
+--------
+
+Before querying over HTTP, you must first enable the HTTP extension in your
+schema. Add this to your schema, outside any ``module``:
+
+.. code-block:: sdl
+
+    using extension edgeql_http;
+
+Then create a new migration and apply it using
+:ref:`ref_cli_edgedb_migration_create` and
+:ref:`ref_cli_edgedb_migrate`, respectively.
+
+Your instance is now able to receive EdgeQL queries over HTTP.
+
+.. note::
+
+    Enabling the HTTP extension is only required for querying over HTTP. It is
+    *not* required for health checks or observability.
+
+Making a query request
+^^^^^^^^^^^^^^^^^^^^^^
+
+Make a query to your EdgeDB database using this URL:
+
+.. code-block::
+
+    http://<hostname>:<port>/db/<database-name>/edgeql
+
+You may make queries via either the POST or GET HTTP method. Query requests can
+take the following fields:
+
+- ``query`` - contains the EdgeQL query string
+- ``variables``- contains a JSON object where the keys are the parameter names
+  from the query and the values are the arguments to be used in this execution
+  of the query.
+
+When using the GET method, supply ``query`` and ``variables`` as query
```

Comment:
maybe add an example of a GET request here? if for no other reason than to show that encoding all of this in the querystring is fiddly and for any non-trivial query they probably want to use POST with a request body